cannon/mipsevm/core/iface.go

```diff
@@ -1,13 +1,21 @@
-package mipsevm
+package core
 
-import "github.com/ethereum/go-ethereum/common"
+import (
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/ethereum-optimism/optimism/cannon/mipsevm/core/debug"
+	"github.com/ethereum-optimism/optimism/cannon/mipsevm/core/memory"
+	"github.com/ethereum-optimism/optimism/cannon/mipsevm/core/witness"
+)
 
 type FPVMState interface {
-	GetMemory() *Memory
+	GetMemory() *memory.Memory
 
 	// GetPC returns the currently executing program counter
 	GetPC() uint32
 
+	GetRegisters() *[32]uint32
```

Comment:
There can be multiple register files used by multiple threads. Which thread's does `patch.go` use?
Could we have `patch.go` instead return the stack pointer (`sp`) and have the caller set that appropriately themselves?

Reply:
I was thinking that this would just return the "active" registers (on the active thread), and similarly for `GetPC`.   I think that works for patching as we should be loading the current program into a single active thread.  

We could refactor as you suggested, but seems a little nicer to just have the patch logic handle it since this would be required any time `PatchStack` is called.  Another option would be to add a `SetStackPointer` method.  WDYT?

Reply:
Gotcha. Yeah it makes sense to have `GetRegisters` refer to the active one then. It's fine as-is. Let's add a godoc to `GetRegisters` highlighting this.